document/static/js/modules/schema/document.js

```diff
@@ -407,15 +407,55 @@ spec.nodes = spec.nodes.update(
         spec.nodes.get("table"),
         {
             attrs: {
-                track: {default: []}
+                track: {default: []},
+                width:{default:"100"},
+                aligned:{default:'center'}
             },
             parseDOM: [{tag: "table", getAttrs(dom) {
                 return {
-                    track: parseTracks(dom.dataset.track)
+                    track: parseTracks(dom.dataset.track),
+                    width: dom.width,
+                    aligned: dom.aligned
                 }
             }}],
             toDOM(node) {
-                const attrs = {}
+                const attrs ={}
+                switch(node.attrs.width){
+                    case "25":
+                        attrs.width = "25",
+                        attrs.class = "table-25"
+                        break;
+                    case "50":
+                        attrs.width = "50",
+                        attrs.class = "table-50"
+                        break;
+                    case "75":
+                        attrs.width = "75",
+                        attrs.class = "table-75"
+                        break;
+                    default:
+                        attrs.width = "100",
+                        attrs.class = "table-100"
+                        break;
+                }
+                if(attrs.width === "100"){
+                    node.attrs.aligned = 'center'
```

Comment:
we shouldn't assign values on the node here. We are meant to translate the node to an HTML entity - not modify it.

Reply:
Fixed in new commit